web/src/lib/settings/migrate.ts

```diff
@@ -0,0 +1,118 @@
+import type { RecursivePartial } from "$lib/types/generic";
+import type { CobaltSettings } from "$lib/types/settings";
+import defaultSettings from "./defaults";
+
+const oldSwitcherValues = {
+    theme: ['auto', 'light', 'dark'],
+    vCodec: ['h264', 'av1', 'vp9'],
+    vQuality: ['720', 'max', '2160', '1440', '1080', '480', '360', '240', '144'],
+    aFormat: ['mp3', 'best', 'ogg', 'wav', 'opus'],
+    filenamePattern: ['classic', 'pretty', 'basic', 'nerdy']
+} as const;
+
+const oldCheckboxes = [
+    'audioMode',
+    'alwaysVisibleButton',
+    'downloadPopup',
+    'fullTikTokAudio',
+    'muteAudio',
+    'reduceTransparency',
+    'disableAnimations',
+    'disableMetadata',
+    'twitterGif',
+    'plausible_ignore',
+    'ytDub',
+    'tiktokH265'
+] as const;
+
+type LegacySwitchers = keyof typeof oldSwitcherValues;
+type LegacyCheckboxes = typeof oldCheckboxes[number];
+
+const _get = (name: LegacyCheckboxes | LegacySwitchers) => {
+    const value = localStorage.getItem(name);
+    if (value !== null) {
+        return value;
+    }
+}
+
+const getBool = (name: LegacyCheckboxes) => {
+    const value = _get(name);
+
+    if (value !== undefined) {
+        return value === 'true';
+    }
+}
+
+const getLiteral = <T extends LegacySwitchers>(name: T) => {
+    const value = _get(name);
+    if (value === undefined) {
+        return;
+    }
+
+    const values = oldSwitcherValues[name] as readonly string[];
+    if (values.includes(value)) {
+        type SwitcherOptions = typeof oldSwitcherValues[T][number];
+        return value as SwitcherOptions;
+    }
+}
+
+const getDownloadMode = () => {
+    if (getBool('muteAudio')) {
+        return 'mute';
+    }
+
+    if (getBool('audioMode')) {
+        return 'audio';
+    }
+
+    return 'auto';
+}
+
+const cleanup = () => {
+    for (const name of [...oldCheckboxes, ...Object.keys(oldSwitcherValues)]) {
```

Comment:
instead of looping through current values, loop through parsed localstorage and remove everything that isn't `settings` or `plausible_ignore`. current implementation does not accommodate for older localstorage values present in fairly ancient versions of cobalt, leaving them behind.